unic/char/property/src/domain.rs

```diff
@@ -16,6 +16,19 @@ use std::fmt::{Debug, Display};
 use std::hash::Hash;
 
 
+/// A Character Property, defined for some or all Unicode characters.
+pub trait CharProperty: PartialCharProperty {
```

Comment:
Every property in [the Property Index](http://unicode.org/reports/tr44/#Property_Index_Table) has an entry in [PropertyAliases.txt](http://www.unicode.org/Public/UCD/latest/ucd/PropertyAliases.txt).

I think we should make `CharProperty` the supertrait and `PartialCharProperty`, `CompleteCharProperty`, `EnumeratedCharProperty`, and `NumericCharProperty` just extend from that.

We can also drop `Partial` as a parent of `Complete` if we do so, which was only done such that there _would_ be a common parent to have as a supertrait of `NumericProperty`/`EnumeratedProperty`.

(Note: I haven't considered Unihan, which has more than the 15 properties that have an entry in PropertyAliases.txt. However, a cursory examination shows that all normative properties have an alias in said document. And we could always just return the same string from both if there's only one.)